src/main/java/com/linkedin/kafka/clients/utils/DefaultHeaderSerializerDeserializer.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright 2017 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License").
+ * See License in the project root for license information.
+ */
+package com.linkedin.kafka.clients.utils;
+import com.linkedin.kafka.clients.consumer.LazyHeaderListMap;
+import java.nio.ByteBuffer;
+import java.util.Map;
+
+
+/**
+ *  <pre>
+ *  8 bytes magic number
+ *  1 byte version/flags
+ *  4 byte length of all headers
+ *  Varies Header key and length fields are encoded in a 4TLV format using network byte ordering.
+ *  </pre>
+ */
+public class DefaultHeaderSerializerDeserializer implements HeaderSerializerDeserializer {
```

Comment:
As mentioned in the other comment. I think we should have separate interface for serializer and deserializer. This class implement both.